File: create_tables/rm_data_store.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table rm_data_store';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table rm_data_store(
        STORE_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        ACTIVE_IND VARCHAR(1),
        ADDRESS_OBS_NO INT,
        ADDRESS_SOURCE VARCHAR(40),
        BUSINESS_ASSOCIATE_ID VARCHAR(40),
        CONTAINED_BY_STORE_ID VARCHAR(40),
        DATA_STORE_HIER_ID VARCHAR(40),
        DATA_STORE_NAME VARCHAR(255),
        DATA_STORE_TYPE VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        HIER_LEVEL_ID VARCHAR(40),
        LOCATION_ID VARCHAR(40),
        LONG_LOCATION VARCHAR(40),
        PHYSICAL_STORE_STATUS VARCHAR(40),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        SHORT_LOCATION VARCHAR(40),
        SOURCE VARCHAR(40),
        TOTAL_CAPACITY INT,
        USED_CAPACITY INT,
        USED_CAPACITY_DATE TIMESTAMP(0),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")
 

File: postgres_39/all_struct_with_default/RmDataStore.go
package dto

type Rm_data_store struct {
	Store_id              string  `json:"store_id" default:"source"`
	Active_ind            *string `json:"active_ind" default:""`
	Address_obs_no        *int    `json:"address_obs_no" default:""`
	Address_source        *string `json:"address_source" default:""`
	Business_associate_id *string `json:"business_associate_id" default:""`
	Contained_by_store_id *string `json:"contained_by_store_id" default:""`
	Data_store_hier_id    *string `json:"data_store_hier_id" default:""`
	Data_store_name       *string `json:"data_store_name" default:""`
	Data_store_type       *string `json:"data_store_type" default:""`
	Effective_date        *string `json:"effective_date" default:""`
	Expiry_date           *string `json:"expiry_date" default:""`
	Hier_level_id         *string `json:"hier_level_id" default:""`
	Location_id           *string `json:"location_id" default:""`
	Long_location         *string `json:"long_location" default:""`
	Physical_store_status *string `json:"physical_store_status" default:""`
	Ppdm_guid             *string `json:"ppdm_guid" default:""`
	Remark                *string `json:"remark" default:""`
	Short_location        *string `json:"short_location" default:""`
	Source                *string `json:"source" default:""`
	Total_capacity        *int    `json:"total_capacity" default:""`
	Used_capacity         *int    `json:"used_capacity" default:""`
	Used_capacity_date    *string `json:"used_capacity_date" default:""`
	Row_changed_by        *string `json:"row_changed_by" default:""`
	Row_changed_date      *string `json:"row_changed_date" default:""`
	Row_created_by        *string `json:"row_created_by" default:""`
	Row_created_date      *string `json:"row_created_date" default:""`
	Row_effective_date    *string `json:"row_effective_date" default:""`
	Row_expiry_date       *string `json:"row_expiry_date" default:""`
	Row_quality           *string `json:"row_quality" default:""`
}
